### PR TITLE
Tests enhancement

### PR DESCRIPTION
**Pull request by fahad-siddiqui**
This pull request has changes to the following files:

1. tests/basic.rs
Added the following test cases for enhanced coverage of range iteration, random operations, and batched writes :
- range_iterate_random()
- range_iterate_overlap()
- range_iterate_sparse_keys()
- get_put_many_random()
- get_put_many_delay()
- get_put_delete_variable_entry()
- override_one_random()
- override_many_random()
- batched_overwrite()
- batched_overwrite_delete()

2. Cargo.toml
Added rand crate and updated the tokio dependency to include the “time” feature.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -26,11 +26,12 @@ tokio-condvar = { version="0.3", features=["parking_lot"] }
 tokio-uring = { version="0.5", optional=true }
 tokio-uring-executor = { package="kioto-uring-executor", version="0.1", optional=true }
 bloomfilter = { version="1", optional=true }
+rand = "0.8.5"
```
Comment (kaimast) [inline Cargo.toml:29]: I think you can move this to "dev-dependencies"

**Comment by kaimast** [comment]
If you run "just fix-formatting" all the tests should pass.

File: Cargo.toml (hunk 1 above)
Comment (fahad-siddiqui) [reply Cargo.toml]: Done. [679eae2](https://github.com/kaimast/lsm-rs/pull/38/commits/679eae253fea2d5aec721ca14ea68671975fdfb5)

**Comment by fahad-siddiqui** [comment]
> If you run "just fix-formatting" all the tests should pass.

Done [c44489d](https://github.com/kaimast/lsm-rs/pull/38/commits/c44489dc83149983e825ba6086d3281e1cdf2683)

**Pull request merged**